tests/surface_input_regions.cpp

```diff
@@ -23,40 +23,43 @@
 
 #include <vector>
 #include <memory>
+#include <experimental/optional>
 
 using namespace testing;
 
-// No tests use subtract as that is not yet implemented in Mir
-
 enum class RegionAction
 {
     add,
     subtract,
 };
 
-struct InputRegion
+struct Region
 {
     struct Element
     {
         RegionAction action;
-        int x, y, width, height;
+        std::pair<int, int> top_left;
+        std::pair<int, int> size;
     };
 
     std::string name;
+    std::pair<int, int> surface_size;
     std::vector<Element> elements;
 
     void apply_to_surface(wlcs::Client& client, wl_surface* surface) const
     {
+        if (elements.empty())
+            return;
         auto const wl_region = wl_compositor_create_region(client.compositor());
```

Comment:
While we're here, does this want to be
`auto const wl_region = wlcs::wrap_wl_object(wl_compositor_create_region(client.compositor());`

Reply:
That could work, but this use case is simple enough to not really need it